base/src/com/thoughtworks/go/util/SystemEnvironment.java

```diff
@@ -166,7 +166,9 @@
     public static GoSystemProperty<String[]> GO_SSL_INCLUDE_PROTOCOLS = new GoStringArraySystemProperty("go.ssl.protocols.include", null);
     public static GoSystemProperty<String[]> GO_SSL_EXCLUDE_PROTOCOLS = new GoStringArraySystemProperty("go.ssl.protocols.exclude", null);
     public static GoSystemProperty<Boolean> GO_SSL_RENEGOTIATION_ALLOWED = new GoBooleanSystemProperty("go.ssl.renegotiation.allowed", true);
-    public static GoSystemProperty<Boolean> GO_CONFIG_REPO_GC_AGGRESSIVE = new GoBooleanSystemProperty("go.config.repo.gc.aggressive", false);
+    public static GoSystemProperty<Boolean> GO_CONFIG_REPO_GC_AGGRESSIVE = new GoBooleanSystemProperty("go.config.repo.gc.aggressive", true);
+    public static GoSystemProperty<Long> GO_CONFIG_REPO_GC_WARNING_THRESHOLD = new GoLongSystemProperty("go.config.repo.gc.warning.threshold", 10000L);
```

Comment:
`GC_LOOSE_OBJECT_THRESHOLD` seems more like upon hitting this threshold a GC would be done. Thats not correct though. The thing is a warning message would be shown when we cross the threshold. 
Agree with other comments.